File: app/src/main/java/com/bopr/android/smailer/BuildInfo.java
package com.bopr.android.smailer;

import android.content.Context;

import java.io.InputStream;
import java.util.Properties;

public class BuildInfo {

    public final String number;
    public final String time;
    public final String name;

    public BuildInfo(Context context) {
        Properties properties = new Properties();
        try {
            try (InputStream stream = context.getAssets().open("release.properties")) {
                properties.load(stream);
                number = properties.getProperty("build_number");
                time = properties.getProperty("build_time");
            }
            name = context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName;
        } catch (Exception x) {
            throw new RuntimeException("Cannot read build info", x);
        }
    }

}

File: app/src/main/kotlin/com/bopr/android/smailer/BuildInfo.kt
package com.bopr.android.smailer

import android.content.Context
import java.io.IOException
import java.util.Properties

/**
 * Application release build info.
 *
 * @author Boris Pronin ([boris280471@gmail.com](mailto:boris280471@gmail.com))
 */
data class BuildInfo(
    val number: String,
    val time: String,
    val name: String?
) {

    companion object {

        fun get(context: Context): BuildInfo {
            try {
                Properties().run {
                    context.assets.open("release.properties").use {
                        load(it)
                    }
                    return BuildInfo(
                        getProperty("build_number"),
                        getProperty("build_time"),
                        context.packageManager.getPackageInfo(context.packageName, 0).versionName
                    )
                }
            } catch (x: IOException) {
                throw RuntimeException("Cannot read release properties file", x)
            }
        }
    }

}